### ISSUE TITLE
Oruga-next with Vue3 doesn't work with up/down keys after searching 

### ISSUE THREAD

**Issue by amir20** [oruga-ui/oruga#249]
### Overview of the problem

**Oruga** version: `@oruga-ui/oruga-next 0.4.8`
**Vuejs** version:  `vue 3.2.22`
**OS/Browser**: Chrome 96

### Description

When using the `autocomplete` component, I am not able to use the up/down keys _after_ searching. 

See video...

https://user-images.githubusercontent.com/260667/143936997-1fcb5081-89c7-4777-9791-738f9fdcba5e.mov

### Steps to reproduce

1. Start typing for items
2. Wait for items to show up in auto complete
3. Use the up and down to move hover

### Expected behavior

There should be a hover item and up and down should move it. 

### Actual behavior

No items are hovered. 

I have tried some time debugging it. The code is correctly calling `setHover` but no css is appended. I am not sure what else to try.

**Comment by jtommy** [comment]
@coreyworrell I'll investigate about it

**Comment by jtommy** [comment]
At the moment it works well but the only one problem is about scrolling options that I just fixed 3461ef489b3f0a5d009f0c0c9d0ae38655fa1b6f; maybe it could related to your issue

**Comment by coreyworrell** [comment]
@jtommy I keep getting pinged in this repo and I'm not sure why :smile:

**Comment by amir20** [comment]
@jtommy what do you mean it works well? Were you able to reproduce the issue?

**Comment by jtommy** [comment]
@amir20 can you reproduce on codepen or similar? So I'll try my fix if are able to reproduce it

**Comment by amir20** [comment]
@jtommy I tried to figure out how to use codepen with orgua-next and vue3. But all the examples in the documentation were with vue2. If you have an example with _any_ component, then I can try again to use the auto complete component. Otherwise, I think it should be easily reproducible. Thanks.

**Comment by jtommy** [comment]
@amir20 https://codesandbox.io/s/silly-sun-u7ron?file=/src/App.vue

It's a sample from Discord server

**Comment by amir20** [comment]
Thanks. It helped. I wasn't able to reproduce my exact issue. Which is confusing. I may need your help. After much debugging, it seems like the object that is passed to `setHover` or `selected` is a proxy. But the data is just a regular array which is why the hover css class never shows up. 

I'll have to keep investigating.

**Comment by jtommy** [comment]
Ok, anyway I fixed a bug about scroll navigation using arrow keys

**Comment by amir20** [comment]
@jtommy I fixed it. To be honest, I have no idea why. Looks I was affected by the proxy equality as explained here in the vue [documentation](https://v3.vuejs.org/guide/reactivity.html#proxy-vs-original-identity). 

This makes no sense. But I wrapped all my objects with `.map((i) => reactive(i))` using vue's reactive function and everything started working. See https://github.com/amir20/dozzle/commit/5cf74e3f959fe5c6950bb672321c14da6aabf259. 

Again no idea why I need to do this. Debugging showed that `this.hovered = {...}` would convert `hovered` to a type `Proxy`.
